src/dft/src/architect/ScanArchitectHeuristic.cpp

```diff
@@ -38,15 +38,84 @@
 #include "ClockDomain.hh"
 #include "ScanArchitect.hh"
 
+#include <boost/geometry.hpp>
+#include <boost/geometry/geometries/register/point.hpp>
+#include <boost/geometry/index/rtree.hpp>
+
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
 namespace dft {
 
 ScanArchitectHeuristic::ScanArchitectHeuristic(
     const ScanArchitectConfig& config,
-    std::unique_ptr<ScanCellsBucket> scan_cells_bucket)
-    : ScanArchitect(config, std::move(scan_cells_bucket))
+    std::unique_ptr<ScanCellsBucket> scan_cells_bucket, utl::Logger* logger)
+    : ScanArchitect(config, std::move(scan_cells_bucket)), logger_(logger)
 {
 }
 
+void ScanArchitectHeuristic::order_reduce_wirelength(std::vector<std::unique_ptr<ScanCell>>& cells)
+{
+
+  // Nothing to order
+  if (cells.empty())
+    return;
+  // No point running this if the cells aren't placed yet
+  if (std::all_of(cells.begin(), cells.end(), [](const std::unique_ptr<ScanCell> &cell) {
+        return !cell->isPlaced();
+      }))
+    return;
+  // Define the starting node as the upperleftmost
+  size_t start_index = 0;
+  int64_t lowest_dist = cells.at(0)->getOrigin().x() + cells.at(0)->getOrigin().y();
```

Comment:
The coordinates of odb allow for negative locations, I don't think code accounts for that with -inf, inf being the upper leftmost point. In practice the coordinates are int_32t so inf = max_int32_t